nostr-client-plus/src/bin/aggregator.rs

```diff
@@ -296,62 +302,76 @@ async fn run() -> Result<()> {
+            let raw_data_id = hex::encode(new_payload.header.raw_data_id.hash());
+
+            // First fetch all existing tasks
+            let saved_tasks: Result<String, RedisError> =
+                (&mut redis_con_task).get(raw_data_id.clone());
+            let mut tasks: Vec<AssignerTask> = match saved_tasks {
+                Ok(val) => {
+                    match from_str(val.as_str()) {
+                        Ok(val) => {
+                            val
+                        }
 
-                        // ToDo: Currently we handle disputes by deleting the entry altogether.
-                        //  If this was already at the second result, another one will come and
-                        //  linger in the book because no other matches are expected to come.
-                        //  Another approach needs to decided.
-                        entry.remove();
-                        PENDING_JOBS.set(aggr_book.len()as i64);
-                        continue;
+                        Err(err) => {
+                            tracing::debug!("Failed parse tasks, error: {}", err);
+                            Vec::new()
+                        }
                     }
+                }
+                Err(err) => {
+                    tracing::debug!("Failed to fetch tasks from redis for data: {}, error: {}", raw_data_id.clone(), err);
+                    Vec::new()
+                }
+            };
 
-                    // Ok, they are the same
-                    workers.push(sender);
-                    // We allow the case of len > N_WORKERS for cases where there are issues
-                    // writing to the db or removing from book.
-                    // The first N_WINNERS are still the same in the DB, regardless.
-
-                    if workers.len() >= n_winners {
-                        tracing::debug!("All the results for {} are consistent, sending", job_id);
-                        let raw_data_id = new_payload.header.raw_data_id.clone();
-                        let db_entry = FinishedJobs {
-                            _id: raw_data_id,
-                            workers: workers.clone(), // if we remove first, no need to clone but this is safer
-                            timestamp: chrono::Utc::now().timestamp() as u64,
-                            result: new_payload,
-                            job_type,
-                            assign_event_id,
-                        };
-                        match collection.insert_one(db_entry, None).await {
+            // Then append new tasks in the end
+            tasks.push(
+                AssignerTask {
+                    worker: sender,
+                    event_id: raw_data_id.clone(),
+                    result: new_payload,
+                }
+            );
+
+            // When number of tasks equal to winner count, let's try to finalize
+            if tasks.iter().len() >= n_winners {
```

Comment:
You don't need to call `iter` to know the length of a vector. `tasks.len()` is enough